src/constants/CacheManager.ts

```diff
@@ -0,0 +1,6 @@
+import { Cache } from '@nestjs/cache-manager';
+
+export class RequiredCacheManager implements Cache {}
+export interface RequiredCacheManager extends Cache {
+  store: Required<Cache['store']>;
+}
```

Comment:
Isso aqui é uma técnica de "type merging", onde você declara uma classe que implementa uma interface, e depois declara uma classe com mesmo nome implementando a mesma interface. Repara que ainda mudei a propriedade store para ser um "required" do tipo que tinha antes. Esse Required faz as propriedades opcionais serem obrigatórias. Vou explicar melhor porque fiz isso mais abaixo.

No entanto, só pra esclarecer: a técnica acima tem q ser usada com cuidado, porque estamos forçando o Typescript a enxergar tipagens que não necessariamente fazem parte do tipo real.